packages/pwa-buildpack/lib/BuildBus/BuildBus.js

```diff
@@ -0,0 +1,123 @@
+const makeDebug = require('debug');
+const Tapable = require('tapable');
+const pertain = require('pertain');
+
+const VALID_HOOKS = new Set(
+    Object.values(Tapable).filter(v => typeof v === 'function')
+);
```

Comment:
Do we want to only support a subset of hooks?

But I suppose `BuildBus` supporting all of the hooks is different from saying Venia will use / declare all of the hooks.
This would allow plugins to support any Tapable hook they wanted, and other plugins to tap into them.

Reply:
@supernova-at This is an early draft API, and in my new suggestion (I'll post it in another comment) it makes a little more sense. I do think we should enforce that you have to use Hooks so that the API is consistent; Tapable's `HookMap` seems to have little utility to me, so I wanted to disallow it.